core/store/src/trie/ops/resharding.rs

```diff
@@ -38,20 +38,58 @@ pub(crate) fn boundary_account_to_intervals(
     retain_mode: RetainMode,
 ) -> Vec<Range<Vec<u8>>> {
     let mut intervals = vec![];
-    // TODO(#12074): generate correct intervals in nibbles.
-    for (col, _) in COLUMNS_WITH_ACCOUNT_ID_IN_KEY {
-        match retain_mode {
-            RetainMode::Left => {
-                intervals.push(vec![col]..[&[col], boundary_account.as_bytes()].concat())
+    for (prefix, name) in col::ALL_COLUMNS_WITH_NAMES {
+        if col::COLUMNS_WITH_ACCOUNT_ID_IN_KEY.contains(&(prefix, name)) {
+            intervals.push(split_column_with_account_id(boundary_account, &retain_mode, prefix));
+            continue;
+        }
+        match prefix {
+            col::DELAYED_RECEIPT_OR_INDICES
+            | col::PROMISE_YIELD_INDICES
+            | col::PROMISE_YIELD_TIMEOUT
+            | col::BANDWIDTH_SCHEDULER_STATE => {
+                intervals.push(copy_column_to_both_children(prefix))
             }
-            RetainMode::Right => {
-                intervals.push([&[col], boundary_account.as_bytes()].concat()..vec![col + 1])
+            col::BUFFERED_RECEIPT_INDICES
+            | col::BUFFERED_RECEIPT
+            | col::BUFFERED_RECEIPT_GROUPS_QUEUE_DATA
+            | col::BUFFERED_RECEIPT_GROUPS_QUEUE_ITEM => {
+                if let Some(interval) = include_column_in_lower_index_child(&retain_mode, prefix) {
+                    intervals.push(interval);
+                }
             }
+            // Poor man's exhaustive check for handling all column types.
+            _ => panic!("Unhandled trie key type: {}", name),
         }
     }
     intervals
 }
 
+fn split_column_with_account_id(
```

Comment:
nit: The name is a bit confusing since this method actually only returns the interval. How about something like the following naming scheme?

* get_interval_for_split
* get_interval_for_copy_to_both
* get_interval_for_copy_to_one

Maybe add `column` or `child`/`children` as needed.

Reply:
Done!